src/reports/reporter.ts

```diff
@@ -11,24 +13,37 @@ export enum EventName {
   START_BINARY = "tabnine-binary-run",
 }
 
+let reporter: TelemetryReporter;
+
 export function initReporter(
   context: ExtensionContext,
   id: string,
   version: string,
   key: string
 ): void {
+  if (inTestMode) return;
+
   reporter = new TelemetryReporter(id, version, key);
   context.subscriptions.push(reporter);
 }
 
 export function report(event: EventName): void {
-  reporter.sendTelemetryEvent(event);
+  if (inTestMode) return;
+
+  void (async () => {
+    const data = await getReportData();
```

Comment:
if any of the `systeminformation` calls  will be rejected, `sendTelemetryEvent` will not be called. is this the expected behavior?

Reply:
I think we should send the event without the properties, thanks